modules/map_preprocess.py

```diff
@@ -0,0 +1,234 @@
+        """
+        Receive incoming depth data
+        """
+        if self._pose_data is None:
+            return
+
+        self.message_queue.queue_message((data, self._pose_data))
+
+    def pose_callback(self, data: data_types.Pose):
+        """
+        Receive incoming pose data
+        """
+        self._pose_data = data
+
+    def process_incoming_data(self):
+        """
+        Process batches of local point clouds
+        """
+        data_in = self.message_queue.wait_for_message()
+
+        if data_in is None:
+            return None
+        
+        data, pose = data_in[1]
+
+        data = self._input_data_preprocess(data)
+        points, count = self._point_data_to_map(data, pose)
+
+        return data_types.MapPreProcessorOut(data_in[0], points, count)
+
+    def _input_data_preprocess(self, data):
+        """
+        Taken the raw data and convert to a standardised (Nx3) matrix of local
+        cartesian cooridantes. This function should be overriden by child class
+        """
+        raise NotImplementedError
+
+    def _point_data_to_map(self, point_cloud, pose):
+        """
+        Take coordinate array and convert to a set of map voxel coordinates and
+        a hit count for each voxel
+        """
+        map_points = self._local_to_global(point_cloud, pose)
+
+        if self.conf.map.enable_compression:
+            map_points, point_count = self._discretise_point_cloud(map_points)
+        else:
+            point_count = 1
+
+        return (map_points, point_count)
+
+    def _initialise_bin_delimitations(self):
+        x_bins = np.linspace(self.map_definition.x_min,
+                             self.map_definition.x_max,
+                             self.map_definition.x_divisions)
+        y_bins = np.linspace(self.map_definition.y_min,
+                             self.map_definition.y_max,
+                             self.map_definition.y_divisions)
+        z_bins = np.linspace(self.map_definition.z_min,
+                             self.map_definition.z_max,
+                             self.map_definition.z_divisions)
+
+        self.bins = (x_bins, y_bins, z_bins)
+
+    def _local_to_global(self, local_points, pose):
+        """
+        Convert local points to a global coordinate system
+        """
+        rot = R.from_quat(pose.quaternion)
+
+        global_points = rot.apply(local_points)
+        global_points = np.add(global_points, pose.translation)
+
+        return global_points
+
+    def _discretise_point_cloud(self, points):
+        """
+        Take continous point cloud and discritise to map grid
+        """
+        x_sort = np.digitize(points[:, 0], self.bins[0]) - 1
+        y_sort = np.digitize(points[:, 1], self.bins[1]) - 1
+        z_sort = np.digitize(points[:, 2], self.bins[2]) - 1
+
+        points = np.column_stack((x_sort, y_sort, z_sort))
+        points = np.uint16(points)
+
+        if self.conf.map.enable_compression:
```

Comment:
Due to the above if statement prevetning this being called with compression, surely this if is never executed?

Reply:
This was a mistake is fixed now